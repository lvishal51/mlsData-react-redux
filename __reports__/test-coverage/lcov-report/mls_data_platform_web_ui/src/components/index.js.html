<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for mls_data_platform_web_ui/src/components/index.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">mls_data_platform_web_ui/src/components</a> index.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>0/0</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">export { default as ZapInput } from './zap-input/ZapInput';
export { default as ZapHeader } from './zap-header/ZapHeader';
export { default as ZapFooter } from './zap-footer/ZapFooter';
export { default as ZapPanel } from './zap-panel/ZapPanel';
export { default as MappedUnmapped } from './zap-mapping/MappedUnmapped';
export { default as CustomPagination } from './zap-pagination/CustomPagination';
export { default as MappedFields } from './zap-mappedFields/MappedFields';
export { default as ContextDataPanel } from './zap-context-panel/ContextDataPanel';
export { default as AutoSuggestion } from './zap-autoSuggestion/AutoSuggestion';
export { default as ZapTable } from './zap-table/ZapTable';
export { default as AutoComplete } from './zap-autoComplete/AutoComplete';
export { default as ZapSideBar } from './zap-sideBar/zapSideBar';
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Thu Nov 29 2018 15:48:26 GMT+0530 (IST)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
