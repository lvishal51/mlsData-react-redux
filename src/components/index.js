export { default as ZapInput } from './zap-input/ZapInput';
export { default as ZapHeader } from './zap-header/ZapHeader';
export { default as ZapFooter } from './zap-footer/ZapFooter';
export { default as ZapPanel } from './zap-panel/ZapPanel';
export { default as MappedUnmapped } from './zap-mapping/MappedUnmapped';
export { default as CustomPagination } from './zap-pagination/CustomPagination';
export { default as MappedFields } from './zap-mappedFields/MappedFields';
export { default as ContextDataPanel } from './zap-context-panel/ContextDataPanel';
export { default as AutoSuggestion } from './zap-autoSuggestion/AutoSuggestion';
export { default as ZapPreview } from './zap-preview/ZapPreview';
export { default as AutoComplete } from './zap-autoComplete/AutoComplete';
export { default as CustomCalendar } from './zap-calendar/CustomCalendar';
export { ZapHeader as ZapHeaderComp } from './zap-header/ZapHeader';
export { ZapPanel as ZapPanelComp } from './zap-panel/ZapPanel';
export { ContextDataPanel as ContextDataPanelComp} from './zap-context-panel/ContextDataPanel';
export { AutoComplete as AutoCompleteComp } from './zap-autoComplete/AutoComplete';
export { MappedFields as MappedFieldsComp} from './zap-mappedFields/MappedFields';
