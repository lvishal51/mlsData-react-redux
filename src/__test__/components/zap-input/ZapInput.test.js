// @flow

import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { ZapInput } from 'components';

const mockFn = jest.fn();
const zapInput = {
  defaultValue: "",
  name: "input",
  placeholder: "type a text",
  type: "text",
  handleChange: mockFn,
  handleBlur: mockFn,
  errorMessage: "Please, write some text",
  minLength: 20,
  maxLength: 40
};

describe('ZapInput component', () => {
  it('ZapInput should render correctly', () => {
    const wrapper = shallow(<ZapInput {...zapInput} />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('ZapInput with label should render correctly', () => {
    const temp = {
      ...zapInput,
      label: "Label"
    }
    const wrapper = shallow(<ZapInput {...temp} />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('ZapInput with label should render correctly', () => {
    const temp = {
      ...zapInput,
      errorState: true,
      errorMessage: "Please enter value"
    }
    const wrapper = shallow(<ZapInput {...temp} />);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('ZapInput OnChange function should be triggered correctly', () => {
    const wrapper = shallow(<ZapInput {...zapInput} />);
    const event = { preventDefault() {}, target: { value: 'theValue' } };
    const spy = jest.spyOn(wrapper.instance(), "handleChange");
    wrapper.instance().forceUpdate();
    wrapper.find("Input").simulate('change', event);
    expect(spy).toBeCalled();
  });

  it('ZapInput OnBlur function should be triggered correctly', () => {
    const wrapper = shallow(<ZapInput {...zapInput} />);
    const event = { preventDefault() {}, target: { value: 'theValue' } };
    const spy = jest.spyOn(wrapper.instance(), "handleBlur");
    wrapper.instance().forceUpdate();
    wrapper.find("Input").simulate('blur', event);
    expect(spy).toBeCalled();
  });

});
