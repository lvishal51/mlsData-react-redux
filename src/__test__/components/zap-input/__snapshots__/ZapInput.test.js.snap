// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ZapInput component ZapInput should render correctly 1`] = `
<div>
  <InputGroup
    tag="div"
  >
    <Input
      className=""
      defaultValue=""
      maxLength={40}
      minLength={20}
      name="input"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="type a text"
      type="text"
    />
  </InputGroup>
</div>
`;

exports[`ZapInput component ZapInput with label should render correctly 1`] = `
<div>
  <label>
    Label
  </label>
  <InputGroup
    tag="div"
  >
    <Input
      className=""
      defaultValue=""
      maxLength={40}
      minLength={20}
      name="input"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="type a text"
      type="text"
    />
  </InputGroup>
</div>
`;

exports[`ZapInput component ZapInput with label should render correctly 2`] = `
<div>
  <InputGroup
    tag="div"
  >
    <Input
      className="inputError"
      defaultValue=""
      maxLength={40}
      minLength={20}
      name="input"
      onBlur={[Function]}
      onChange={[Function]}
      placeholder="type a text"
      type="text"
    />
  </InputGroup>
  <p
    className="errorMessageText"
  >
    Please enter value
  </p>
</div>
`;
